packages/office-addin-dev-settings/src/commands.ts

```diff
@@ -26,28 +27,32 @@ export async function appcontainer(manifestPath: string, command: commander.Comm
           const allowed = await ensureLoopbackIsEnabled(manifestPath, askForConfirmation);
           console.log(allowed ? "Loopback is allowed." : "Loopback is not allowed.");
         } catch (err) {
-          throw new Error(`Unable to allow loopback for the appcontainer. \n${err}`);
+          throw new ExpectedError(`Unable to allow loopback for the appcontainer. \n${err}`);
         }
       } else if (command.preventLoopback) {
         try {
           const name = await getAppcontainerNameFromManifestPath(manifestPath);
           await removeLoopbackExemptionForAppcontainer(name);
           console.log(`Loopback is no longer allowed.`);
         } catch (err) {
-          throw new Error(`Unable to disallow loopback. \n${err}`);
+          throw new ExpectedError(`Unable to disallow loopback. \n${err}`);
         }
       } else {
         try {
           const name = await getAppcontainerNameFromManifestPath(manifestPath);
           const allowed = await isLoopbackExemptionForAppcontainer(name);
           console.log(allowed ? "Loopback is allowed." : "Loopback is not allowed.");
         } catch (err) {
-          throw new Error(`Unable to determine if appcontainer allows loopback. \n${err}`);
+          throw new ExpectedError(`Unable to determine if appcontainer allows loopback. \n${err}`);
         }
       }
-      usageDataObject.sendUsageDataSuccessEvent("appcontainer");
+      usageDataObject.reportSuccess("appcontainer");
     } catch (err) {
-      usageDataObject.sendUsageDataException("appcontainer", err);
+      if (err instanceof ExpectedError){
+        usageDataObject.reportExpectedError("appcontainer", err);
```

Comment:
perhaps we should change usageDataObject to be:

usageDataObject.reportError("appcontainer", err) with an optional isExpected parameter which could be explicitly set or would be implicity set based on whether it is ExpectedError. So there could be different usage. Here it can just pass err but for other usage it could be:

// this is an expected error
usageDataObject.reportError("eventName", err, true);

Reply:
I've set the function name to be reportException, because a reportError function already existed and I didn't want to change its name

Reply:
Let's chat. There are a few things we should clarify. You're doing great. I am thinking that we need to understand better how the underlying telemetry data can distinguish the three cases (success, expected error, unexpected error). I think we may want to set Succeeded = true even for expected errors, so this is only false for unexpected errors. But then for all successful telemetry (when Succeeded == true), how would we tell between success and expected errors. Do we have ExpectedError = true? Perhaps.

I know currently you have Succeeded = true for success, and false for both expected and unexpected errors, and then Expected = true for expected errors. 

The only real reason to change this is that it aligns more with how other telemetry is done.

Reply:
Makes sense. It was like that before, but I changed it to put together with the report exception function and that changed these.
The way I am thinking of making the change you suggested is putting an ```if``` in the beginning of the ```reportException``` to check if the error is an expected error. If that's the case, we treat it as succeded as needed.

I like this way because the user only calls a single function (```reportException```) for all possible errros, but in the back, we are checking and treating the cases differently.